db/repos/dapps.js

```diff
@@ -22,87 +22,84 @@ var PQ = require('pg-promise').ParameterizedQuery;
  * @param {Database} db - Instance of database object from pg-promise
  * @param {Object} pgp - pg-promise instance to utilize helpers
  * @constructor
- * @return {DappsRepo}
+ * @return {DappsRepository}
  */
-function DappsRepo (db, pgp) {
-	this.db = db;
-	this.pgp = pgp;
+class DappsRepository {
 
-	this.sortFields = ['name'];
-}
+	constructor (db, pgp) {
+		this.db = db;
+		this.pgp = pgp;
+
+		// TODO: A proper repository shouldn't need to export any properties like this:
+		this.sortFields = ['name'];
+	}
+
+	/**
+	 * Counts dapps by transaction id
+	 * @param {string} id
+	 * @return {Promise<number>}
+	 */
+	countByTransactionId (id) {
+		return this.db.one(sql.countByTransactionId, id, a => +a.count);
```

Comment:
The usage of the code is expecting a row object with a count attribute.  This change will break the code. 

https://github.com/LiskHQ/lisk/blob/cc2e8998efe14b428316ae8affef135bbc6e0658/logic/inTransfer.js#L83-L85

I believe the change you did is good, but you have to update the usage of this repository method as well.

Reply:
I have updated all usage cases as well ;)